File: Android/AceRFID/app/src/main/java/dngsoftware/acerfid/Filament.java
package dngsoftware.acerfid;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "filament_table")
public class Filament {

        @SuppressWarnings("UnusedDeclaration")
        @PrimaryKey(autoGenerate = true)
        public int dbKey;

        @SuppressWarnings("UnusedDeclaration")
        @ColumnInfo(name = "filament_position")
        public int position;

        @SuppressWarnings("UnusedDeclaration")
        @ColumnInfo(name = "filament_name")
        public String filamentName;

        @SuppressWarnings("UnusedDeclaration")
        @ColumnInfo(name = "filament_id")
        public String filamentID;

        @SuppressWarnings("UnusedDeclaration")
        @ColumnInfo(name = "filament_vendor")
        public String filamentVendor;

        @SuppressWarnings("UnusedDeclaration")
        @ColumnInfo(name = "filament_param")
        public String filamentParam;

}

File: Android/AceRFID/app/src/main/java/dngsoftware/acerfid/Filament.kt
package dngsoftware.acerfid

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "filaments")
data class Filament(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,

    @ColumnInfo(name = "filament_position")
    var position: Int = 0,

    @ColumnInfo(name = "filament_name")
    var filamentName: String? = null,

    @ColumnInfo(name = "filament_id")
    var filamentID: String? = null,

    @ColumnInfo(name = "filament_vendor")
    var filamentVendor: String? = null,

    @ColumnInfo(name = "filament_param")
    var filamentParam: String? = null,
)